### PR TITLE
Metric Minimization

### PR DESCRIPTION
**Pull request by lydiaxing**
This PR adds the objective field to the metric object for metric minimization.

Not I also updated the `maven-compiler-plugin` to be version 1.7 instead of 1.6 because otherwise the test cases did not run.

### CODE REVIEW

File: src/test/java/com/sigopt/model/ExperimentTest.java [hunk 1]
```diff
@@ -72,6 +72,7 @@ public void constructFromJson() throws Exception {
         assertEquals(200.0, exp.getMetrics().get(0).getValueBaseline(), 1e-9);
         assertEquals("Sales", exp.getMetrics().get(1).getName());
         assertNull(exp.getMetrics().get(1).getValueBaseline());
+        assertEquals(exp.getMetrics().get(1).getObjective(), "maximize");
```
Comment (pfhayes) [inline src/test/java/com/sigopt/model/ExperimentTest.java]: Expected case should come first to match the other tests

File: pom.xml [hunk 2]
```diff
@@ -105,8 +105,8 @@
                 <artifactId>maven-compiler-plugin</artifactId>
                 <version>3.3</version>
                 <configuration>
-                    <source>1.6</source>
-                    <target>1.6</target>
+                    <source>1.7</source>
+                    <target>1.7</target>
```
Comment (pfhayes) [inline pom.xml]: seems fine but unrelated to this change? did your change necessitate this, or were you unable to run the tests even on master?

Comment (lydiaxing) [reply pom.xml]: Yeah I couldn't run tests on master either got this error:
Screen Shot 2019-07-02 at 2 47 03 PM

Comment (lydiaxing) [reply pom.xml]: I'm going to break this out into another PR

File: src/main/java/com/sigopt/model/Metric.java [hunk 3]
```diff
@@ -10,6 +10,11 @@ public Metric(String name) {
         this.set("name", name);
     }
 
+    public Metric(String name, String objective) {
+        super();
+        this.set("objective", objective);
+    }
+
```
Comment (oliviakim95) [inline src/main/java/com/sigopt/model/Metric.java]: hmm do we need this? Looks like we don't have one for `value_baseline`

Comment (lydiaxing) [reply src/main/java/com/sigopt/model/Metric.java]: Good point, thank you will remove!

**Pull request merged**